pkg/client/cache/reflector.go

```diff
@@ -88,6 +100,34 @@ func NewReflector(lw ListerWatcher, expectedType interface{}, store Store, resyn
 	return r
 }
 
+// getDefaultReflectorName walks back through the call stack until we find a caller from outside of the ignoredPackages
+// it returns back a shortpath/filename:line to aid in identification of this reflector when it starts logging
+func getDefaultReflectorName(ignoredPackages ...string) string {
+	name := "????"
+outer:
+	for i := 1; i < 10; i++ {
+		_, file, line, ok := goruntime.Caller(i)
+		if !ok {
+			break
+		}
+		for _, ignoredPackage := range ignoredPackages {
+			if strings.Contains(file, ignoredPackage) {
+				continue outer
+			}
+
+		}
+
+		pkgLocation := strings.LastIndex(file, "/pkg/")
```

Comment:
> Does this impose a requirement that consuming code/projects are structured with the pkg/ convention?

No, it just makes the string prettier if you do.  If you do, it gives everything past `pkg` for shorter display.  Otherwise, it logs with the entire long package.